product/scanning.py

```diff
@@ -116,40 +116,172 @@ def _retrieve_cached_scan_result(
     ):
         return self.retrieve_scan_result(container_image=container_image, component=component)
 
-    def upload_image(
-            self,
-            container_image: ContainerImage,
-            component: Component,
-            wait_for_result: bool=False
-        ):
-        metadata = self._metadata(container_image=container_image, component=component)
+    def process_image(
+        self,
+        container_image: ContainerImage,
+        component: Component,
+        processing_mode: ProcessingMode,
+        wait_for_result: bool=False,
+    ):
+        if processing_mode is ProcessingMode.FORCE_UPLOAD:
+            # Shortcut: Upload image no matter what
+            default_custom_data = self._metadata(
+                container_image=container_image,
+                component=component,
+            )
+            processing_result = self.upload_image_to_protecode(
```

Comment:
why are there two methods: 'upload_image_to_protecode' and 'upload_image'?

also, I would prefer to separate the logic determining which action to take and the logic that actually takes the action.

in particular, I would very much prefer to not mix the introduction of new functionality (rescan) with refactoring/restructuring